### PR TITLE
v1.2 go

### PR DESCRIPTION
**Pull request by ksyeo1010**
I will need to release the package once the binding part is approved. Can't link to commit since URL changed from previous package.

### CODE REVIEW

File: binding/go/pv_recorder.go [hunk 1]
```diff
@@ -83,34 +83,32 @@ func pvRecorderStatusToString(status PvRecorderStatus) string {
...
 // PvRecorder struct
 type PvRecorder struct {
-	// handle for pvrecorder instance in C.
+	// handle for pvRecorder instance in C.
 	handle uintptr
 
-	// Index of audio device to start recording and capture audio.
-	DeviceIndex int
-
 	// FrameLength to get for each read command.
```
Comment (laves) [inline binding/go/pv_recorder.go:89]: Length of each frame returned from read command.

File: binding/go/pv_recorder.go [hunk 2]
```diff
@@ -157,31 +155,42 @@ func (pvrecorder *PvRecorder) Stop() error {
...
 }
 
+// SetDebugLogging enables or disables debug logging for PvRecorder. Debug logs will indicate when there are
+// overflows in the internal  frame buffer and when an audio source is generating frames of silence.
+func (pvRecorder *PvRecorder) SetDebugLogging(isDebugLoggingEnabled bool) {
+	nativePvRecorder.nativeSetDebugLogging(pvRecorder, isDebugLoggingEnabled)
+}
+
+// GetIsRecording gets whether the given instance is currently recording audio or not.
+func (pvRecorder *PvRecorder) GetIsRecording() bool {
```
Comment (laves) [inline binding/go/pv_recorder.go:183]: I would just use this more as a property name (like we did with .NET and Python)
```suggestion
func (pvRecorder *PvRecorder) IsRecording() bool {
```

File: demo/go/README.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# PvRecorder Go Demo
```
Comment (laves) [inline demo/go/README.md:1]: Pedantic, I know, but the consistency will look good
```suggestion
# PvRecorder Demo for Go
```

File: demo/go/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+## Usage
+
+NOTE: The working directory for the following go commands is:
+
+```console
+pvrecorder/demo/go
+```
+
+### Demo
```
Comment (laves) [inline demo/go/README.md:17]:
```suggestion

```

File: README.md [hunk 5]
```diff
@@ -100,6 +112,63 @@ Hit `Ctrl+C` to stop recording. If no audio device index (`-d`) is provided, the
...
+To start recording, initialize the instance and run start function:
+
+```go
+import . "github.com/Picovoice/pvrecorder/binding/go"
+
+recorder := PvRecorder{
+    FrameLength: 512,
+    DeviceIndex: -1, // Using -1 for index uses default audio input device.
+    BufferedFramesCount: 10,
+}
```
Comment (laves) [inline README.md:141]: I think we need a `NewPvRecorder` constructor that gives us an instance with default args. We do this with a bunch of our engines and it's a pattern in Go.

**Review by laves — CHANGES_REQUESTED** [review]

File: binding/go/README.md [hunk 6]
```diff
@@ -1,75 +1,70 @@
...
-    FrameLength: 512,
-    BufferSizeMSec: 1000,
-    LogOverflow: 1,
-}
-if err := recorder.Init(); err != nil {
-    // error
+devices = GetAvailableDevices() // select index of device
+
+recorder = NewPvRecorder(/*FrameLength*/512)
+recorder.DeviceIndex = 0 //
```
Comment (laves) [inline binding/go/README.md:44]: empty comment

File: binding/go/README.md [hunk 7]
```diff
@@ -1,75 +1,70 @@
...
+recorder.Init()
+if err != nil {
+// handle init error
```
Comment (laves) [inline binding/go/README.md:47]: indent

File: binding/go/README.md [hunk 8]
```diff
@@ -1,75 +1,70 @@
...
 }
-if err := recorder.Start(); err != nil {
-    // error
+defer recorder.Delete()
+
+err = recorder.Start()
+if err != nil {
+// handle start error
```
Comment (laves) [inline binding/go/README.md:53]: indent

**Review by laves — CHANGES_REQUESTED** [review]
spelling and demos are failing. I'm guessing you're waiting on release to let demo actions pass?

**Comment by ksyeo1010** [comment]
> spelling and demos are failing. I'm guessing you're waiting on release to let demo actions pass?

yes

**Review by laves — APPROVED** [review]

**Pull request merged**